mautic/mautic
# Date lead field fixes

## Description
There are few ways how to get a badly formatted date to the database and then Mautic throws an error while it tries to load it to a Date(Time) object. Reported in https://github.com/mautic/mautic/issues/1181

## Steps to reproduce
1. Create a date lead field with default value 20/07/1981
2. Try to edit a lead.
You should get an error.

## Steps to test
Apply this PR and test again. Firstly, this PR adds a validation to the default value of datetime, date and time lead fields. Please test if the validation works for each of these fields. It must work when creating a new field as well as when modifying an old one. Also, you should get a date/time picker depending on the field type.

Also, if a lead has some date/time with unknown value format in it, the lead detail should ignore such value and shouldn't end with an error.
